docs/blog/2019-02-20-introducing-use-static-query/index.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Introducing useStaticQuery
+date: 2019-02-20
+author: Sidhartha Chatterjee
+tags:
+  - hooks
+  - static query
+---
+
+React Hooks are _cool_. Besides simplifying your code and removing the need for a lot of boilerplate associated with classes in JavaScript (looking at you, `this`), they enable some serious shareability. They also make it possible to use state in functional components.
+
+You can probably tell that we've been super excited about [React Hooks][hooks-intro]. So when they finally landed in React 16.8, we figured it was time to give our very own `StaticQuery` component the hook treatment!
+
+### Say hello to useStaticQuery
+
+`useStaticQuery` is a hook that takes a GraphQL query and returns your data. That's it.
+
+**No more Render Props necessary to use a Static Query**
+
+This simplifies accessing data in your components and also keeps your component tree shallow.
+
+Let's quickly check out a simple example. Here's a simple Header component, first using `StaticQuery` and then with `useStaticQuery`.
+
+### Before
+
+```jsx
+import React from "react"
+import { StaticQuery, graphql } from "gatsby"
+
+const Header = () => {
+  return (
+    <StaticQuery
+      query={graphql`
+        query {
+          site {
+            siteMetadata {
+              title
+            }
+          }
+        }
+      `}
+      render={data => (
+        <header>
+          <h1>{data.site.siteMetadata.title}</h1>
+        </header>
+      )}
+    />
+  )
+}
+
+export default Header
+```
+
+### After
+
+```jsx
+import React from "react"
+import { useStaticQuery, graphql } from "gatsby"
+
+const Header = () => {
+  const { site } = useStaticQuery(
+    graphql`
+      query {
+        site {
+          siteMetadata {
+            title
+          }
+        }
+      }
+    `
+  )
+
+  return (
+    <header>
+      <h1>{site.siteMetadata.title}</h1>
+    </header>
+  )
+}
+
+export default Header
+```
+
+Isn't that cleaner and more succinct?
+
+Just like `StaticQuery`, when you build your site for production, Gatsby will parse _and_ execute your queries in `useStaticQuery` and inject data in just the right place.
+
+Where this gets even more powerful is the ability to create custom hooks that use `useStaticQuery` in them. Let's say you need to query for the site title several times in your app. Instead of a call to `useStaticQuery` in each component, you can extract it out to a custom hook like:
+
+```jsx
+const useSiteMetadata = () => {
+  const { site } = useStaticQuery(
+    graphql`
+      query {
+        site {
+          siteMetadata {
+            title
+          }
+        }
+      }
+    `
+  )
+  return site.siteMetadata
+}
+```
+
+Now everytime you need the site title, you can use this custom hook. And in case you change the structure of that data, you only need to change the query once!
```

Comment:
Can you also show an example of this being consumed? I think that would help!

Reply:
Great catch! Done.